pkg/policy/policies.go

```diff
@@ -224,6 +286,572 @@ func (ps *Policies) FilterableInUserlandMap() map[*Policy]int {
+	}
+
+	opts := &bpf.BPFMapCreateOpts{
+		BTFFD:                 uint32(btfFD),
+		BTFKeyTypeID:          info.BTFKeyTypeID,
+		BTFValueTypeID:        info.BTFValueTypeID,
+		BTFVmlinuxValueTypeID: info.BTFVmlinuxValueTypeID,
+		MapFlags:              info.MapFlags,
+		MapExtra:              info.MapExtra,
+		MapIfIndex:            info.IfIndex,
+	}
+	newInnerMap, err := bpf.CreateMap(
+		prototypeMap.Type(),
+		fmt.Sprintf("%s_%d", mapName, mapVersion), // new map name
+		prototypeMap.KeySize(),
+		prototypeMap.ValueSize(),
+		int(prototypeMap.MaxEntries()),
+		opts,
+	)
+	if err != nil {
+		return nil, errfmt.WrapError(err)
+	}
+
+	return newInnerMap, nil
+}
+
+func updateOuterMap(m *bpf.Module, mapName string, mapVersion uint16, innerMap *bpf.BPFMapLow) error {
+	outerMap, err := m.GetMap(mapName)
+	if err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	u32Key := uint32(mapVersion)
+	keyPointer := unsafe.Pointer(&u32Key)
+	innerMapFD := uint32(innerMap.FileDescriptor())
+	valuePointer := unsafe.Pointer(&innerMapFD)
+
+	// update version filter map
+	// - key is the map version
+	// - value is the related filter map FD.
+	if err := outerMap.Update(keyPointer, valuePointer); err != nil {
+		return errfmt.WrapError(err)
+	}
+
+	return nil
+}
+
+func (ps *Policies) createNewFilterMapsVersion(bpfModule *bpf.Module) error {
+	mapsNames := map[string]string{ // inner map name: outer map name
+		UIDFilterMap:      UIDFilterMapVersion,
+		PIDFilterMap:      PIDFilterMapVersion,
+		MntNSFilterMap:    MntNSFilterMapVersion,
+		PidNSFilterMap:    PidNSFilterMapVersion,
+		UTSFilterMap:      UTSFilterMapVersion,
+		CommFilterMap:     CommFilterMapVersion,
+		CgroupIdFilterMap: CgroupIdFilterVersion,
+		BinaryFilterMap:   BinaryFilterMapVersion,
+	}
+
+	polsVersion := ps.Version()
+	for innerMapName, outerMapName := range mapsNames {
+		// TODO: This only spawns new inner filter maps. Their termination must
+		// be tackled by the versioning mechanism.
+		newInnerMap, err := createNewInnerMap(bpfModule, innerMapName, polsVersion)
+		if err != nil {
+			return errfmt.WrapError(err)
+		}
+
+		// outerMap maps:
+		// 1. uid_filter_version       u32, uid_filter
+		// 2. pid_filter_version       u32, pid_filter
+		// 3. mnt_ns_filter_version    u32, mnt_ns_filter
+		// 4. pid_ns_filter_version    u32, pid_ns_filter
+		// 5. cgroup_id_filter_version u32, cgroup_id_filter
+		// 6. uts_ns_filter_version    u32, uts_ns_filter
+		// 7. comm_filter_version      u32, comm_filter
+		// 8. binary_filter_version    u32, binary_filter
+		if err := updateOuterMap(bpfModule, outerMapName, polsVersion, newInnerMap); err != nil {
+			return errfmt.WrapError(err)
+		}
+
+		// store pointer to the new inner map version
+		ps.bpfInnerMaps[innerMapName] = newInnerMap
+	}
+
+	return nil
+}
+
+func (ps *Policies) createNewEventsMapVersion(bpfModule *bpf.Module, eventsState map[events.ID]events.EventState) error {
+	// Initialize events parameter types map
```

Comment:
Events parameters should be constant and I don't see a reason for it to be changed between policy updates - maybe move to init so we won't repeat this logic on every policy update?

Reply:
Yep, they're constant indeed. We need here only to tackle eventStates. Going to move eventsParams to initialization.